### PR TITLE
Front-End Scaff with React, Redux

### PR DESCRIPTION
**Pull request by lazdinst**
tasks: [
  {
    id: 16,
    title: 'Setup Front-end Scaffold'
  },
  {
    id: 22,
    title: 'Add Redux to Front-End Scaffolding'
  }
]

### CODE REVIEW

**Review by Sam0064 — APPROVED** [review]

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import Landing from './main/Landing.jsx';
+import Header from './header/Header.jsx';
+import Footer from './footer/Footer.jsx';
+import RootRouter from './main/RootRouter.jsx';
+
+const App = () => {
+  //TEMP DATA
```
Comment (olegtrofimov1) [inline client/src/components/App.jsx:8]: Please delete all commented out code

File: client/src/components/footer/Footer.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Grid, Row, Col } from 'react-bootstrap';
+
+// TODO: Breakout into custom CSS
```
Comment (olegtrofimov1) [inline client/src/components/footer/Footer.jsx:4]: Good use of TODO tag here

File: client/src/components/main/Home.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import PostList from './PostList.jsx';
+
+//Define 
+const NO_SELECT_MSG = 'No Selection';
+const NO_INPUT_MSG = 'No Message';
+
+const Home = (props) => {
+  let playGround = false;
+
+  if (playGround) {
```
Comment (olegtrofimov1) [inline client/src/components/main/Home.jsx:30]: This looks like code that was copy and pasted from a tutorial or another app. Please remove everything that is not relevant specifically to your end product.

File: client/src/components/main/PostList.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { Grid, Row, Col, Panel } from 'react-bootstrap';
+import PostListEntry from './PostListEntry.jsx';
+
+const PostList = (props) => {
+  //TEMP DATA
+  //let data = [{title: 'First Post', postMemory: 'This is my first memory'}];
```
Comment (olegtrofimov1) [inline client/src/components/main/PostList.jsx:9]: Please remove commented out  temp data.

File: client/src/components/main/PostList.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  //let posts = data.map((post, i) => { return <PostListEntry post={post} key={i} />; });
+
+  return (
+    <Panel collapsible defaultExpanded header="Panel Header" bsStyle="info">
+      This is where posts go
+    </Panel>
+  );
+};
+
+//export default PostList;
```
Comment (olegtrofimov1) [inline client/src/components/main/PostList.jsx:19]: Don't we need to use PostList elsewhere? Why are you commenting out the export here?

File: client/src/store/index.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { createStore, applyMiddleware, compose } from 'redux';
+import { routerMiddleware } from 'react-router-redux';
+import thunk from 'redux-thunk';
+import createHistory from 'history/createBrowserHistory';
+import rootReducer from './modules';
+
+/**
```
Comment (olegtrofimov1) [inline client/src/store/index.js:7]: Great job with clear comments in this file 👍

File: db/DB_README.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 ## Schema Creation / Modification
 To create new table:
 1. create new model file for table in models folder
-2. run 'knex migrate:make "SCHEMA_CHANGE_ACTION_NAME"' on command line
+2. run 'knex migrate:make "SCHEMA_CHANGE_ACTION_NAME"' on command line; this wil generate the migration file name
```
Comment (olegtrofimov1) [inline db/DB_README.md:4]: spelling

**Review by olegtrofimov1 — COMMENTED** [review]
This PR is way too big (+678!!!). There are so many things happening here and it is hard to review thoroughly. 

There are so many things happening here. You could break this single PR into like 4-5 different ones. 1) Simple App/Landing/Header/Footer 2) Add React Router 3) Add Redux  4) Add Map component 5) Allow ability to create post

You have code in here that looks like its not related to the actual app (something part of Redux playground?). 

Also there is not a single test in here.

Overall this is something you want to avoid. A huge pull request results in people not reviewing every line carefully and just accepting this code. This results in a bunch of unnecessary or unused code in the code base. Also this creates a huge amount of tech debt (you are going to have to delete that code and write those tests eventually).

**Pull request merged**